crates/bevy_input/src/input.rs

```diff
@@ -31,42 +58,64 @@ where
         self.pressed.insert(input);
     }
 
+    /// Check if `input` has been pressed.
     pub fn pressed(&self, input: T) -> bool {
         self.pressed.contains(&input)
     }
 
+    /// Register a release for input `input`.
     pub fn release(&mut self, input: T) {
         self.pressed.remove(&input);
         self.just_released.insert(input);
     }
 
+    /// Check if `input` has been just pressed.
     pub fn just_pressed(&self, input: T) -> bool {
         self.just_pressed.contains(&input)
     }
 
+    /// Check if `input` has been just pressed, and clear it so that it won't be processed further.
+    pub fn just_pressed_and_clear(&mut self, input: T) -> bool {
+        self.just_pressed.remove(&input);
+        self.just_pressed.contains(&input)
```

Comment:
This will always return false because we just removed input in the previous line. I'll just fix this now because its easy.

Reply:
Sometime I really miss the 🤦‍♂️ emoji reaction on GitHub... sorry about that one

Reply:
Haha no worries. This is a class of mistake that we all make :smile: